custom_components/goldair_climate/device.py

```diff
@@ -198,11 +210,13 @@ def _retry_on_failed_connection(self, func, error_message):
         for i in range(self._CONNECTION_ATTEMPTS):
             try:
                 func()
+                self._api_protocol_working = True
                 break
             except Exception as e:
                 _LOGGER.debug(f"Retrying after exception {e}")
                 if i + 1 == self._CONNECTION_ATTEMPTS:
                     self._reset_cached_state()
+                    self._api_protocol_working = False
```

Comment:
Thanks, I was being a bit lazy with the previous logic :)

Reply:
After running overnight, this seems to be the only part of the change that has had a significant impact.  Avoiding switching protocols definitely has improved the availability of my protocol 3.3 device, overnight it was always available, compared with frequent disconnects before I updated.